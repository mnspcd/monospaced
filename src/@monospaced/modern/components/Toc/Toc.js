import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router";

import "./Toc.css";

/**
 * Use `Toc` to display lists of links in a Table of Contents format.
 */
const Toc = ({ children }) => <ol className="Toc">{children}</ol>;

Toc.propTypes = {
  /**
   * Toc.Item components to display in the Toc.
   */
  children: PropTypes.node.isRequired,
};

Toc.Item = ({ date, slug, title }) => {
  const LinkElement = slug ? Link : "a";

  return (
    <li className="Toc-item">
      <LinkElement
        className="Toc-itemLink"
        href={slug ? undefined : "#"}
        to={slug && `/${slug}/`}
      >
        <span className="Toc-itemTitle">{title}</span>
        {date && <span className="Toc-itemDate">{date}</span>}
      </LinkElement>
    </li>
  );
};

Toc.Item.displayName = "Toc.Item";

Toc.Item.propTypes = {
  date: PropTypes.string,
  slug: PropTypes.string,
  title: PropTypes.string.isRequired,
};

export default Toc;
