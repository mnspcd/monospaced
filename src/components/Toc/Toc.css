.Toc {
  border-top: calc(1rem / 16);
  font-family: var(--font-monospaced);
  font-variant: slashed-zero;
  list-style: none;
  margin-bottom: 0;
  margin-top: 0;
  padding-left: 0;
}

.Toc-itemLink {
  display: flex;
  justify-content: space-between;
}
