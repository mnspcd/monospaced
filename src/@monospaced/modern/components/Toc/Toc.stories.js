import { number, text } from "@storybook/addon-knobs";
import React from "react";

import Toc from "./Toc";

export default {
  component: Toc,
  title: "Components/Toc",
};

export const basic = () => (
  <Toc>
    <Toc.Item date="2018-06-11" slug="" title="Item-1" />
    <Toc.Item date="2018-06-12" slug="" title="Item-2" />
    <Toc.Item date="2018-06-13" slug="" title="Item-3" />
  </Toc>
);

export const knobs = () => {
  const numberOfItems = number(
    "number of items",
    4,
    {
      range: true,
      min: 1,
      max: 20,
    },
    "Story",
  );

  return (
    <Toc>
      {[...Array(numberOfItems).keys()].map(i => (
        <Toc.Item
          date={`${text("date", "2018-06-", "Component")}${i + 11}`}
          key={`item-${i + 1}`}
          title={`${text("title", "Item", "Component")}-${i + 1}`}
        />
      ))}
    </Toc>
  );
};

knobs.story = { parameters: { docs: { disable: true } } };
