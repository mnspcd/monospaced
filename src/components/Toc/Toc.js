import { Link } from "react-router";
import PropTypes from "prop-types";
import React from "react";

import "./Toc.css";

const Toc = ({ children }) => <ol className="Toc">{children}</ol>;

Toc.propTypes = {
  children: PropTypes.node.isRequired,
};

Toc.Item = ({ date, slug, title }) => {
  const LinkElement = slug ? Link : "a";

  return (
    <li className="Toc-item">
      <LinkElement
        className="Toc-itemLink"
        href={!slug ? "#" : undefined}
        to={slug && `/${slug}/`}
      >
        <span className="Toc-itemTitle">{title}</span>
        {date && <span className="Toc-itemDate">{date}</span>}
      </LinkElement>
    </li>
  );
};

Toc.Item.propTypes = {
  date: PropTypes.string,
  slug: PropTypes.string,
  title: PropTypes.string.isRequired,
};

export default Toc;
