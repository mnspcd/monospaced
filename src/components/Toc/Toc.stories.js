import React from "react";
import { number, text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Toc from "./Toc";

storiesOf("Components/Toc", module).add("Component", () => {
  const numberOfItems = number(
    "number of items",
    4,
    {
      range: true,
      min: 1,
      max: 20,
    },
    "Story",
  );

  return (
    <Toc>
      {[...Array(numberOfItems).keys()].map(i => (
        <Toc.Item
          date={`${text("date", "2018-06-", "Component")}${i + 11}`}
          key={`item-${i + 1}`}
          title={`${text("title", "Item", "Component")}-${i + 1}`}
        />
      ))}
    </Toc>
  );
});
