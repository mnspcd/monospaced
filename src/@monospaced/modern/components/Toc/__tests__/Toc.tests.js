/* eslint-env jest */
import { render } from "@testing-library/react";
import React from "react";

import Toc from "..";

describe("Toc component", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Toc>
        <Toc.Item title="title" />
      </Toc>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render date correctly", () => {
    const { container } = render(
      <Toc>
        <Toc.Item date="2018-11-15" title="title" />
      </Toc>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render slug correctly", () => {
    const { container } = render(
      <Toc>
        <Toc.Item slug="path" title="title" />
      </Toc>,
    );

    expect(container).toMatchSnapshot();
  });
});
