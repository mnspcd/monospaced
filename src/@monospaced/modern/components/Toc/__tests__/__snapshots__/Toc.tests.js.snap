// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toc component should render correctly 1`] = `
<div>
  <ol
    class="Toc"
  >
    <li
      class="Toc-item"
    >
      <a
        class="Toc-itemLink"
        href="#"
      >
        <span
          class="Toc-itemTitle"
        >
          title
        </span>
      </a>
    </li>
  </ol>
</div>
`;

exports[`Toc component should render date correctly 1`] = `
<div>
  <ol
    class="Toc"
  >
    <li
      class="Toc-item"
    >
      <a
        class="Toc-itemLink"
        href="#"
      >
        <span
          class="Toc-itemTitle"
        >
          title
        </span>
        <span
          class="Toc-itemDate"
        >
          2018-11-15
        </span>
      </a>
    </li>
  </ol>
</div>
`;

exports[`Toc component should render slug correctly 1`] = `
<div>
  <ol
    class="Toc"
  >
    <li
      class="Toc-item"
    >
      <a
        class="Toc-itemLink"
      >
        <span
          class="Toc-itemTitle"
        >
          title
        </span>
      </a>
    </li>
  </ol>
</div>
`;
